.github/workflows/main.yml

```diff
@@ -276,10 +276,8 @@ jobs:
           set -x
           docker run --rm -i --name libvcx --network host $DOCKER_IMG_NAME_LIBVCX \
                               bash -c '(cd $HOME/libvcx && \
-                                RUST_TEST_THREADS=1 TEST_POOL_IP=127.0.0.1 cargo test --release && \
-                                RUST_TEST_THREADS=1 TEST_POOL_IP=127.0.0.1 cargo test --release --features "pool_tests" && \
                                 RUST_TEST_THREADS=1 TEST_POOL_IP=127.0.0.1 cargo test --release --features "agency_v2" && \
```

Comment:
Thanks for heads up on that, I tried one more time, something little bit different (on your list I think it would fall into `option1`)
```
RUST_TEST_THREADS=1 TEST_POOL_IP=127.0.0.1 cargo test --release --features "aries" && \
RUST_TEST_THREADS=1 TEST_POOL_IP=127.0.0.1 cargo test --release --features "pool_agency_tests pool_tests"
```
- So the first time all cheap tests are executed
- Second time around, pool tests and `pool_agency_tests` is executed
Whereas I replaced the feature combo of "agency2 + pool" with `pool_agency_tests` "feature", as it was the single tests using such combination.

Reply:
I agree with this approach for now, but would prefer to move in the direction of option 2. during the refactor process.

Reply:
I agree. Also I've split the testing in 2 jobs. It's ci code duplication but it reduces testing time by about 35 minutes

Reply:
>  Use features on all tests and filter accordingly

Actually I am going to take the this option forward